### PR TITLE
Edit pr-data.csv 

### PR DESCRIPTION
**Pull request by Rette66**
For the following test, I found that the project has been moved to Gradle so I moved these tests to gr-data.csv and changed the status in pr-data.csv:

1. org.springframework.boot.actuate.autoconfigure.endpoint.web.documentation.BeansEndpointDocumentationTests.beans
2. org.springframework.boot.context.properties.bind.CollectionBinderTests.bindToSetShouldNotAllowDuplicateValues
3. org.springframework.boot.diagnostics.analyzer.BeanCurrentlyInCreationFailureAnalyzerTests.cycleWithAutowiredFields
4. org.springframework.boot.diagnostics.analyzer.BeanCurrentlyInCreationFailureAnalyzerTests.cycleWithAutowiredFields

### CODE REVIEW

File: format_checker/pr_checker.py [hunk 1]
```diff
@@ -51,6 +51,7 @@
         "Skipped",
         "MovedOrRenamed",
         "Claimed",
+        "MovedToGradle"
```
Comment (darko-marinov) [inline format_checker/pr_checker.py:54]: Should there be a comma at the end of the line, as the previous `Claimed` seems to have had a comma at the end?

File: gr-data.csv [hunk 2]
```diff
@@ -51,6 +51,10 @@ https://github.com/romalev/reactive-spring-web-flux-service,19e64d356840720f78b5
 https://github.com/romalev/reactive-spring-web-flux-service,19e64d356840720f78b50bd4c08cf5a1d7391834,.,com.reactive.service.ApplicationTester.verifyMultiplyOperationOnSuccess,ID,,,please run this project using NonDex v1.2
 https://github.com/romalev/reactive-spring-web-flux-service,19e64d356840720f78b50bd4c08cf5a1d7391834,.,com.reactive.service.ApplicationTester.verifyPowerOf2SubSqrtOperationOnSuccess,ID,,,please run this project using NonDex v1.2
 https://github.com/romalev/reactive-spring-web-flux-service,19e64d356840720f78b50bd4c08cf5a1d7391834,.,com.reactive.service.ApplicationTester.verifySubOperationOnSuccess,ID,,,please run this project using NonDex v1.2
+https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd1414ee4d4,spring-boot-project/spring-boot-actuator-autoconfigure,org.springframework.boot.actuate.autoconfigure.endpoint.web.documentation.BeansEndpointDocumentationTests.beans,ID,,,
```
Comment (darko-marinov) [inline gr-data.csv:54]: Put here the commit SHA that you actually ran, not the old commit from when the project was using Maven. Are the modules correct? Are the test names correct?

File: pr-data.csv [hunk 3]
```diff
@@ -4684,8 +4684,8 @@ https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd14
...
 https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd1414ee4d4,spring-boot-project/spring-boot-autoconfigure,org.springframework.boot.autoconfigure.websocket.servlet.WebSocketMessagingAutoConfigurationTests.basicMessagingWithJsonResponse,ID,InspiredAFix,https://github.com/spring-projects/spring-boot/pull/30514,
-https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd1414ee4d4,spring-boot-project/spring-boot,org.springframework.boot.context.properties.bind.CollectionBinderTests.bindToSetShouldNotAllowDuplicateValues,ID,,,
-https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd1414ee4d4,spring-boot-project/spring-boot,org.springframework.boot.diagnostics.analyzer.BeanCurrentlyInCreationFailureAnalyzerTests.cycleWithAutowiredFields,ID,,,
+https://github.com/spring-projects/spring-boot,daa3d457b71896a758995c264977bdd1414ee4d4,spring-boot-project/spring-boot,org.springframework.boot.context.properties.bind.CollectionBinderTests.bindToSetShouldNotAllowDuplicateValues,ID,MovedToGradle,,
```
Comment (darko-marinov) [inline pr-data.csv:4687]: It'd be nice (but not necessary) to know when the project (or even more precisely the module/test) moved from Maven to Gradle.

File: readme.md [hunk 4]
```diff
@@ -144,6 +144,7 @@ MovedOrRenamed | For test that has a different fully-qualified name on two diffe
 RepoArchived | For test that is in an archived repo, which is indicated by GitHub in messages such as "This repository has been archived by the owner. It is now read-only."
 Deprecated | For test that is in a deprecated repository, which is usually indicated in the project README or description as "Deprecated" or a similar message. To use this status, please also provide some [Notes](#adding-notes) that contain a link to the commit that marks the repository as deprecated. 
 RepoDeleted | For tests that are in repository that does not exist anymore and the link to the repository throws a 404 status code error.
+MovedToGradle | For tests that are in repository that moved from Java Maven to Java Gradle.
```
Comment (darko-marinov) [inline readme.md:147]: You can remove `Java` from `Java Maven` and `Java Gradle`.

**Review by darko-marinov — CHANGES_REQUESTED** [review]
Great contribution. Please address minor issues, definitely list the right commit for Gradle, and maybe find where the project moved from Maven to Gradle.

File: pr-data.csv (hunk 3 above)
Comment (Rette66) [reply pr-data.csv:4687]: I read through Wiki and all release notes, and I believe they migrated to Gradle in about 2021. There are about 46000 commits, and I cannot find the exact commit they migrated from Maven to Gradle.

Comment (darko-marinov) [reply pr-data.csv:4687]: Please see my email.

**Comment by kaiyaok2** [comment]
Hi there - I ran these tests using the NonDex Gradle plugin (with 50 NonDex runs in full mode), and found that `org.springframework.boot.actuate.autoconfigure.endpoint.web.documentation.BeansEndpointDocumentationTests.beans` and `org.springframework.boot.context.properties.bind.CollectionBinderTests.bindToSetShouldNotAllowDuplicateValues` seem to be **not** flaky. These tests might be fixed by developers, or maybe their dependencies were once flaky but got fixed. I would suggest 

On the other hand, `org.springframework.boot.diagnostics.analyzer.BeanCurrentlyInCreationFailureAnalyzerTests.cycleWithAutowiredFields` and `org.springframework.boot.test.autoconfigure.properties.AnnotationsPropertySourceTests.propertiesFromMultipleMetaAnnotationsAreMappedUsingTheirOwnPropertyMapping` seem to be true positive ID tests.

Thanks for your contribution!

**Comment by darko-marinov** [comment]
For the two tests that are not ID, please find where exactly they stopped being ID, before or after the move to Gradle.

**Comment by darko-marinov** [comment]
Are you planning to finish this PR, or should I ask someone else to do it?

**Comment by Rette66** [comment]
Hi Professor Darko, Sorry for not working on this PR last week. I had a really busy week last week. I will try to find the date they moved from Maven to Gradle by tomorrow night. I’m learning how to search changes in Git. Best, Wenxiu  从 Windows 版邮件发送 发件人: darko-marinov发送时间: 2023年10月16日 17:34收件人: TestingResearchIllinois/idoft抄送: Werfero; State change主题: Re: [TestingResearchIllinois/idoft] Move some tests from pr-data.csv to gr-data.csv (PR #793) Are you planning to finish this PR, or should I ask someone else to do it?—Reply to this email directly, view it on GitHub, or unsubscribe.You are receiving this because you modified the open/close state.Message ID: ***@***.***>

**Review by darko-marinov — CHANGES_REQUESTED** [review]
Sort the tests as checker requires (the latest build failed), and consider whether to have this PR only change the Maven part and another PR add to `gr-data.csv`.

**Pull request merged**